services/auth/source/oauth2/init.go

```diff
@@ -30,6 +30,11 @@ const ProviderHeaderKey = "gitea-oauth2-provider"
 
 // Init initializes the oauth source
 func Init(ctx context.Context) error {
+	// if oauth is disabled, we don't need to initialize anything
+	if !setting.OAuth2.Enable {
```

Comment:
According to the CI, this should be called
```suggestion
	if !setting.OAuth2.Enabled {
```

Reply:
Oops, sorry about that.  Looks like commit d9b38494 renamed `setting.OAuth2.Enable` to `setting.OAuth2.Enabled`, but I wrote my patch for 1.21.11 which is before that change.
Can we get the patch merged to the release/v1.21 as-is, and separately tweak it to work for main too?

Reply:
There are three strange things about your request:
1. We typically merge changes the other way around (main -> backport) and our automation is intended for that use as well
2. This PR targets main
3. 1.21. is no longer supported: at the moment, we can barely support versions one number down (so 1.22.x here), two versions is completely out of the picture currently.

Reply:
As far as I can tell 1.22 is the most recent version, is it not?  Anyway, I appreciate that supporting multiple versions can be difficult, and I don't need much "support", I'd just like to get a fix applied to that branch to minimize private patches we're maintaining.

@techknowlogick, could you fix this PR to work for main, and then perhaps spin a new one for the 1.21 branch?  (I'm somewhat constrained by $DAYJOB policy so I can't directly do that myself)